playkit/src/main/java/com/kaltura/playkit/PlayerLoader.java

```diff
@@ -249,4 +249,15 @@ public void addListener(Object groupId, Enum type, PKEvent.Listener listener) {
     public void removeListeners(@NonNull Object groupId) {
         messageBus.removeListeners(groupId);
     }
+
+    @NonNull
+    @Override
+    public <PluginType> List<PluginType> getLoadedPluginsOfType(Class<PluginType> pluginClass) {
+        List<PluginType> filteredPlugins = new ArrayList<>();
+        for (LoadedPlugin loadedPlugin : loadedPlugins.values()) {
+            if (pluginClass.isAssignableFrom(loadedPlugin.plugin.getClass()))
+                filteredPlugins.add((PluginType) loadedPlugin.plugin);
+        }
```

Comment:
Why did you change from my suggested implementation? I'm not saying it's not correct, just want to know if the other one wasn't good.

Reply:
`isInstance()` checks only a superclass. But in case of `PKMediaEntryInterceptor `, which is interface, we have to use `isAssignableFrom`, because it checks both superclass and superinterface.

Reply:
Ok, makes sense. Is there a warning when you cast `loadedPlugin.plugin` to `PluginType`?

Reply:
yes, `Unchecked cast`

Reply:
So please suppress this warning on the statement, and add text that explains it (like I did).